mempool/monitor.go

```diff
@@ -386,3 +438,80 @@ func (p *MempoolMonitor) CollectAndStore() error {
 
 	return nil
 }
+
+// UnconfirmedTxnsForAddress indexes (1) outpoints in mempool that pay to the
+// given address, (2) previous outpoint being consumed that paid to the address,
+// and (3) all relevant transactions. See txhelpers.AddressOutpoints for more
+// information. The number of unconfirmed transactions is also returned. This
+// satisfies the rpcutils.MempoolAddressChecker interface for MempoolMonitor.
+func (p *MempoolMonitor) UnconfirmedTxnsForAddress(address string) (*txhelpers.AddressOutpoints, int64, error) {
+	p.addrMap.mtx.Lock()
+	defer p.addrMap.mtx.Unlock()
+	addrStore := p.addrMap.store
+	if addrStore == nil {
+		return nil, 0, fmt.Errorf("uininitialized MempoolAddressStore")
+	}
+
+	// Retrieve the AddressOutpoints for this address.
+	outs := addrStore[address]
```

Comment:
The Mutex is really guarding both the values and the map itself. Since the `TxnsStore` fields of the `txhelpers.MempoolAddressStore` values are potentially modified, it needed to be locked.  Perhaps the mapped `AddressOutpoints` values could have their own lock, but I'm not sure it's needed.